error message could explain `execute` vs `load`

<!--
If you want to report a bug, we added some points below you can fill out. If you want to request a feature, feel free to remove all the irrelevant text. In addition to this [issue tracker](https://github.com/diesel-rs/diesel/issues), you can also talk to Diesel maintainers and users on [Gitter](https://gitter.im/diesel-rs/diesel).
-->

## Setup

### Versions

- **Rust:** nightly 1.21 2017-08-29
- **Diesel:** 0.14
- **Database:** postgresql 9.5
- **Operating System** ubuntu 16.04

### Feature Flags

- **diesel:** postgres, uuid, serde_json, chrono
- **diesel_codegen:** postgres

## Problem Description

It took me a long time to figure out that `execute` returns at most one value (I'm assuming) while `load` returns (or can return) a collection of values. 

I got the following error message: 

```
error[E0308]: mismatched types
  --> src/main.rs:95:9
   |
95 | /         sql::<(Integer, pg_types::Uuid,)>(query)
96 | |             .execute(&db)
97 | |             .expect("query failed");
   | |_______________________________________^ expected struct `std::vec::Vec`, found usize
   |
   = note: expected type `std::vec::Vec<(i32, uuid::Uuid)>`
              found type `usize`
```
(You can't see it in the error message, but I was trying to assign the query result to a `Vec<(i32, uuid::Uuid)>`).

It would be helpful if this error message could offer some hint about the difference between `execute` vs `load`. Just thought I would share a point of confusion from someone trying to learn the library. Thanks!

## Checklist

- [ yes ] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.


<!--
Thank you for your submission!  You're helping make Diesel more robust 🎉  

We'll try to respond as quickly as possible.
-->

--- a/diesel/src/query_dsl/mod.rs
+++ b/diesel/src/query_dsl/mod.rs
@@ -875,8 +875,12 @@ impl<T: Table> QueryDsl for T {}
 pub trait RunQueryDsl<Conn>: Sized {
     /// Executes the given command, returning the number of rows affected.
     ///
-    /// Used in conjunction with [`insert_into`](../fn.insert_into.html),
-    /// [`update`](../fn.update.html) and [`delete`](../fn.delete.html)
+    /// `execute` is usually used in conjunction with [`insert_into`](../fn.insert_into.html),
+    /// [`update`](../fn.update.html) and [`delete`](../fn.delete.html) where the number of
+    /// affected rows is often enough information.
+    ///
+    /// When asking the database to return data from a query, [`load`](fn.load.html) should
+    /// probably be used instead.
     ///
     /// # Example
     ///
@@ -922,8 +926,12 @@ pub trait RunQueryDsl<Conn>: Sized {
     /// When this method is called on [`sql_query`],
     /// the return type can only be a struct which implements [`QueryableByName`]
     ///
+    /// For insert, update, and delete operations where only a count of affected is needed,
+    /// [`execute`] should be used instead.
+    ///
     /// [`Queryable`]: ../deserialize/trait.Queryable.html
     /// [`QueryableByName`]: ../deserialize/trait.QueryableByName.html
+    /// [`execute`]: fn.execute.html
     /// [`sql_query`]: ../fn.sql_query.html
     ///
     /// # Examples

